src/pages/index.js

```diff
@@ -1,75 +1,117 @@
-const profileDescriptionInput = document.querySelector(
-  "#profile-description-input"
-);
-const profileEditForm = profileEditModal.querySelector(".modal__form");
+import PopUpWithConfirmation from "../components/PopUpWithConfirmation.js";
 
 const cardTemplate =
   document.querySelector("#card-template").content.firstElementChild;
 const cardListEl = document.querySelector(".cards__list");
 
-/* ------------------------------------------------------------------------------- */
-
-/* Add Buttons */
-const addCardButton = document.querySelector(".profile__add-button");
-const addCardForm = document.querySelector(".modal__card-form");
-const addNewCardModal = document.querySelector("#add-card-modal");
-
-/* ------------------------------------------------------------------------------- */
-
-/* Preview Elements */
-const previewImage = document.querySelector(".modal__preview-image");
-const previewDescription = document.querySelector(
-  ".modal__preview-description"
-);
-
 const cardSelector = "#card-template";
 
-/* ------------------------------------------------------------------------------- */
-
 const userInfo = new UserInfo({
   profileTitle: ".profile__title",
   profileDescription: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
-/* ------------------------------------------------------------------------------- */
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "b0f5599c-4007-4941-89b3-30100c4f8838",
+    "Content-Type": "application/json",
+  },
+});
 
-/* Functions */
+let section;
 
-addCardButton.addEventListener("click", () => {
-  addModal.open();
-});
+Promise.all([api.getInitialCards(), api.getUserInfo()])
+  .then(([cards, data]) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: (data) => {
+          const cardEl = renderCard(data);
+          section.addItem(cardEl);
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
 
-/* ------------------------------------------------------------------------------- */
+    userInfo.setUserInfo({
+      title: data.name,
+      description: data.about,
+    });
+
+    userInfo.setAvatar({ avatar: data.avatar });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+if (profileAddButton) {
+  profileAddButton.addEventListener("click", () => {
+    addModal.open();
+  });
+} else {
+  console.error("profileAddButton not found in the DOM");
+}
 
 /* Event Listener */
 profileEditButton.addEventListener("click", () => {
   profileTitleInput.value = userInfo.getUserInfo().title;
-  profileDescriptionInput.value = userInfo.getUserInfo().title;
+  profileDescriptionInput.value = userInfo.getUserInfo().description;
   editModal.open();
 });
-// add new card
 
 const profileEditFormValidator = new FormValidator(config, profileEditForm);
 const addCardFormValidator = new FormValidator(config, addCardForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```